File: articles/the-self-in-jungian-thought-totality-center-and-symbol/go/self_integration_scores.go
package main

import (
	"fmt"
	"math"
)

func totalityScore(
	egoCoherence,
	unconsciousActivation,
	symbolicCenterStrength,
	relationalCoordination,
	disjunction,
	inflationRisk float64,
) float64 {
	return 0.48*egoCoherence +
		0.36*unconsciousActivation +
		0.58*symbolicCenterStrength +
		0.54*relationalCoordination -
		0.46*disjunction -
		0.30*math.Max(inflationRisk, 0)
}

func selfRelationIndex(
	totality,
	differentiation,
	symbolicCenterStrength,
	inflationRisk float64,
) float64 {
	return 0.40*totality +
		0.34*differentiation +
		0.28*symbolicCenterStrength -
		0.30*math.Max(inflationRisk, 0)
}

func main() {
	totality := totalityScore(0.74, 0.66, 0.82, 0.76, 0.28, 0.18)
	selfRelation := selfRelationIndex(totality, 0.80, 0.82, 0.18)

	fmt.Printf("Synthetic totality score: %.3f\n", totality)
	fmt.Printf("Synthetic Self-relation index: %.3f\n", selfRelation)
}

File: articles/the-self-in-jungian-thought-totality-center-and-symbol/rust/self_integration_scores.rs
fn totality_score(
    ego_coherence: f64,
    unconscious_activation: f64,
    symbolic_center_strength: f64,
    relational_coordination: f64,
    disjunction: f64,
    inflation_risk: f64,
) -> f64 {
    0.48 * ego_coherence
        + 0.36 * unconscious_activation
        + 0.58 * symbolic_center_strength
        + 0.54 * relational_coordination
        - 0.46 * disjunction
        - 0.30 * inflation_risk.max(0.0)
}

fn self_relation_index(
    totality: f64,
    differentiation: f64,
    symbolic_center_strength: f64,
    inflation_risk: f64,
) -> f64 {
    0.40 * totality
        + 0.34 * differentiation
        + 0.28 * symbolic_center_strength
        - 0.30 * inflation_risk.max(0.0)
}

fn main() {
    let totality = totality_score(0.74, 0.66, 0.82, 0.76, 0.28, 0.18);
    let self_relation = self_relation_index(totality, 0.80, 0.82, 0.18);

    println!("Synthetic totality score: {:.3}", totality);
    println!("Synthetic Self-relation index: {:.3}", self_relation);
}
